### ISSUE TITLE
Changing TypeReference on Inspector gives an error

### ISSUE THREAD

**Issue by bilal-arikan** [lordofduct/spacepuppy-unity-framework-3.0#1]
Hi,
when i try to change TypeReference value from editor, it gives an error.

My Code:
```C#
[TypeReference.Config(typeof(MonoBehaviour))]
public TypeReference type;
```

Error:

InvalidOperationException: Operation is not valid due to the current state of the object
System.Collections.Stack.Pop () (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Collections/Stack.cs:329)
UnityEngine.GUILayoutUtility.EndLayoutGroup () (at C:/buildslave/unity/build/Runtime/IMGUI/Managed/GUILayoutUtility.cs:323)
UnityEngine.GUILayout.EndVertical () (at C:/buildslave/unity/build/Runtime/IMGUI/Managed/GUILayout.cs:342)
UnityEditor.HostView.OldOnGUI () (at C:/buildslave/unity/build/Editor/Mono/HostView.cs:114)
UnityEngine.Experimental.UIElements.IMGUIContainer.DoOnGUI (UnityEngine.Event evt) (at C:/buildslave/unity/build/Runtime/UIElements/Managed/IMGUIContainer.cs:182)

**Comment by lordofduct** [comment]
Hrmm...

So I was able to recreate. The actual callstack isn't something I directly access, it's something going on on Unity. But I must be doing something that triggers that.

I'll have to do more testing to track down where that is, since the call stack doesn't directly reference what I'm doing.

Currently at this time the error doesn't stop the functionality of TypeReference or Unity. It just spits out in the console. This will be lower priority, but I'll get to when I have the chance.

**Comment by bilal-arikan** [comment]
Yes, looks like there is no problem on the functionality, type value can be changed. 
Thanks for the answer in short time,
